src/zelos/tools/zdbserver/zdbserver.py

```diff
@@ -0,0 +1,340 @@
+
+from xmlrpc.server import SimpleXMLRPCServer
+
+import zelos
+
+
+DEFAULT_INTERFACE = "localhost"
+DEFAULT_PORT = 62433
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger("zdbserver")
+logger.setLevel(logging.DEBUG)
+
+
+class ZdbServer:
+    """
+    Zelos Remote Debug Server.
+
+    Provides remote access to a target running in `zelos` over `xmlrpc`.
+    Numbers are represented as strings for arguments and return values,
+    e.g. '0x1234' or '4660' to support 64-bit numbers, which are
+    otherwise not serialized by `xmlrpc`.
+
+    The `zdbserver` can be run from the command-line via
+    `python -m zelos.tools.zdbserver ZELOS_ARGUMENTS`.
+    """
+
+    def __init__(self, z: zelos.Zelos):
+        self.z = z
+
+    def read_memory(self, address: str, nbytes: int) -> xmlrpc.client.Binary:
+        """
+        Reads specified region of memory.
+
+        Args:
+            address: Address to start reading from.
+            nbytes: Number of bytes to read.
+
+        Returns:
+            Binary data corresponding to data held in memory.
+        """
+        address = int(address, 0)
+        logger.debug(f"[debug] read_memory(0x{address:x}, 0x{nbytes:x})")
+        data = self.z.memory.read(address, nbytes)
+        return xmlrpc.client.Binary(data)
+
+    def write_memory(self, address: str, value: xmlrpc.client.Binary) -> bool:
+        """
+        Writes specified bytes to memory.
+
+        Args:
+            address: Address to start writing data to.
+            value: Binary data to write in memory.
+
+        Returns:
+            True if no exception occurred.
+        """
+        address = int(address, 0)
+        logger.debug(f"[debug] write_memory(0x{address:x}, {value.data})")
+        self.z.memory.write(address, value.data)
+        return True
+
+    def read_register(self, register: str) -> str:
+        """
+        Reads specified register from of the current thread.
+
+        Args:
+            register: String representation of the register name.
+
+        Returns:
+            Value of the specified register.
+        """
+        logger.debug(f"[debug] read_register('{register}')")
+        register = register.lower()
+        if register == "pc":
+            return "0x{0:x}".format(self.z.regs.getIP())
+        return "0x{0:x}".format(self.z.regs[register])
+
+    def write_register(self, register: str, value: str) -> bool:
+        """
+        Writes specified register of the current thread.
+
+        Args:
+            register: String representation of the register name.
+            value: register value to set.
+
+        Returns:
+            True if no exception occurred.
+        """
+        logger.debug(f"[debug] write_register('{register}', {value})")
+        value = int(value, 0)
+        register = register.lower()
+        if register == "pc":
+            self.z.regs.setIP(value)
+        else:
+            self.z.regs[register] = value
+        return True
```

Comment:
Similarly to the above comment, why not return the numeric value written to the register? To be consistent with the `set_reg` function & related apis in the `RegsApi`

Reply:
done